### PR TITLE
WIP: Update notebooks to work with most recent version of padasip

### PR DESCRIPTION
**Pull request by pckbls**
Hi there,

I'm currently taking a course at university on adaptive filter design and I came across padasip. Thank you very much for taking the time to create both padasip and especially those notebooks, they help me a lot learning the fundamentals behind the topic! The notebooks however do not work with the most recent version of padasip. I've attempted to fix that. Unfortunately I've encountered two issues.

1. In `padasip_adaptive_filters_basics.ipynb` when running the NLMS filter cell the resulting coefficients are different from yours.

2. In `AR_identification.ipynb` I cannot execute cell 5 (the one that shows different learning rate curves):
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-10-27ac6cb03956> in <module>()
     12     errors_w, mu_range = filters[i]["filter"].explore_learning(d, x, mu_start=filters[i]["mu_s"],
     13                                         mu_end=filters[i]["mu_e"],
---> 14                                         steps=100, ntrain=0.5, epochs=1, criteria="MSE", target_w=h)
     15 
     16     # save the best learning rate for later use

/home/pckbls/.local/lib/python3.6/site-packages/padasip/filters/base_filter.py in explore_learning(self, d, x, mu_start, mu_end, steps, ntrain, epochs, criteria, target_w)
    163             y, e, w = self.pretrained_run(d, x, ntrain=ntrain, epochs=epochs)
    164             if type(target_w) != bool:
--> 165                 errors[i] = get_mean_error(w-target_w, function=criteria)
    166             else:
    167                 errors[i] = get_mean_error(e, function=criteria)

/home/pckbls/.local/lib/python3.6/site-packages/padasip/misc/error_evaluation.py in get_mean_error(x1, x2, function)
    238     """
    239     if function == "MSE":
--> 240         return MSE(x1, x2)
    241     elif function == "MAE":
    242         return MAE(x1, x2)

/home/pckbls/.local/lib/python3.6/site-packages/padasip/misc/error_evaluation.py in MSE(x1, x2)
    194     """
    195     e = get_valid_error(x1, x2)
--> 196     return np.dot(e, e) / float(len(e))
    197 
    198 def RMSE(x1, x2=-1):

ValueError: shapes (1498,4) and (1498,4) not aligned: 4 (dim 1) != 1498 (dim 0)

<matplotlib.figure.Figure at 0x7fb039e9eba8>
```

Do you have an idea of how to fix those two issues?

Thanks in advance,
Patrick

### CODE REVIEW

File: notebooks/padasip_adaptive_filters_basics.ipynb [hunk 1]
```diff
@@ -288,7 +277,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "And the resulting coefficients are: [ 1.93581967  0.08784035 -3.8879132   0.57809913]\n"
+      "And the resulting coefficients are: [ 1.86906475  0.04664808 -3.88837325  0.46195532]\n"
```
Comment (pckbls) [inline notebooks/padasip_adaptive_filters_basics.ipynb:280]: The resulting coefficients are different.

**Comment by matousc89** [comment]
Thanks for all informations, time and work. I will fix everything as soon
as I will get to computer (middle of the next week).

Dne 28.7.2017 22:19 napsal uživatel "Patrick Bartels" <
notifications@github.com>:

**Comment by matousc89** [comment]
I released new version of Padasip that should fix the issue. Please check it out.

**Comment by pckbls** [comment]
Hi there, sorry for the late response, I'm quite low on time right now. I've upgraded padasip, the notebook now works. Thanks for the fix!

**Pull request merged**